src/net_processing.cpp

```diff
@@ -1087,25 +1087,25 @@ void PeerManagerImpl::PushNodeVersion(CNode& pnode, int64_t nTime)
     // Note that pnode->GetLocalServices() is a reflection of the local
     // services we were offering when the CNode object was created for this
     // peer.
-    ServiceFlags nLocalNodeServices = pnode.GetLocalServices();
+    uint64_t my_services{pnode.GetLocalServices()};
     uint64_t nonce = pnode.GetLocalNonce();
     const int nNodeStartingHeight{m_best_height};
     NodeId nodeid = pnode.GetId();
     CAddress addr = pnode.addr;
 
-    CAddress addrYou = addr.IsRoutable() && !IsProxy(addr) && addr.IsAddrV1Compatible() ?
-                           addr :
-                           CAddress(CService(), addr.nServices);
-    CAddress addrMe = CAddress(CService(), nLocalNodeServices);
+    CService addr_you = addr.IsRoutable() && !IsProxy(addr) && addr.IsAddrV1Compatible() ? addr : CService();
+    uint64_t your_services{addr.nServices};
```

Comment:
nit, can be const
```diff
-    uint64_t my_services{pnode.GetLocalServices()};
-    uint64_t nonce = pnode.GetLocalNonce();
+    const uint64_t my_services{pnode.GetLocalServices()};
+    const uint64_t nonce{pnode.GetLocalNonce()};
     const int nNodeStartingHeight{m_best_height};
-    NodeId nodeid = pnode.GetId();
+    const NodeId nodeid{pnode.GetId()};
     CAddress addr = pnode.addr;
 
-    CService addr_you = addr.IsRoutable() && !IsProxy(addr) && addr.IsAddrV1Compatible() ? addr : CService();
-    uint64_t your_services{addr.nServices};
+    const CService addr_you{addr.IsRoutable() && !IsProxy(addr) && addr.IsAddrV1Compatible() ? addr : CService()};
+    const uint64_t your_services{addr.nServices};
```

Reply:
Will include this if I have to retouch for other reasons.